design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,176 @@
+1. Site administrator: manages the k8s Cluster and is responsible for deploying the ClusterLink into the cluster.
+
+Before deploying the ClusterLink, a few prerequisite steps should be completed:
+
+1. The fabric administrator should create the site certificates (public and private) and transfer them, along with the public CA certificate, to the site administrator.
+2. The site administrator should deploy the certificates as k8s secrets to the cluster.
+3. The site administrator should deploy the ClusterLink operator to the cluster and register ClusterLink Custom Resource Definition (CRD) class to the cluster.
+
+After completing all the prerequisites, the site administrator can deploy the ClusterLink CRD instance to the operator. The operator will then create the ClusterLink components, including:
+
+* A ClusterLink dedicated namespace, which by default will be cl-operator-ns.
+* Deployment of ClusterLink controlplane components, including controlplane-pod, controlplane-service, and RBAC roles.
+* Deployment of ClusterLink dataplane components, including dataplane pods (single or multiple) and the dataplane-service.
+* Deployment of ClusterLink ingress, providing an external access point using load-balancer/node-port/gateway-API service."
+
+Overall, the ClusterLink deployment stages are:
+
+<img src="deployment.png" width="800" height="400" alt="ClusterLink Deployment Stages"/>
+
+The ClusterLink operator will have privileged permissions, allowing it to create a dedicated namespace and the ClusterLink components within.
+The ClusterLink components will be created in `clusterlink-system-ns`, and the control-plane will have privileged permissions within the namespace.
+Additionally, the operator will create and update components in response to the CRD instance create or update actions. Once created or updated, the operator will not actively monitor the state of each component, assuming that any changes will be made only by privileged users.
+Furthermore, the operator will delete all components and the namespace of ClusterLink when the CRD instance is deleted.
+The ClusterLink deployment is limited to one instance per cluster. In the case of deploying more than one CRD instance, only the first one will be taken into account, and the others will be ignored.
+
+The deployment file for the operator and example of CRD instance will be included in every project release.
+
+The ClusterLink CLI will be utilized to automate the deployment process.
+For instance, the following commands can assist the fabric administrator in creating the site certificates:
+
+    clusterlink create fabric --name <fabric_name>
+    clusterlink create site --name <site_name> --fabric <fabric_name> 
+
+For the site administrator, automation of deploying the certificates as a secret to the cluster, deploying the ClusterLink operator, and the CRD instance can be achieved with:
+
+    clusterlink install --site <site_name>
```

Comment:
This is a good point to provide the certificates and keys needed as input and deploy them as secrets. You should consider if these are provided by the CRD or as parameters here.
Since the CRD does not reference secret objects, I assume their names are fixed. Thus we would want to automate their creation so the user is not assigning different names by mistake.